src/API/Management/EmailTemplates.php

```diff
@@ -60,18 +107,49 @@ public function patch($templateName, $data)
      * See docs @link below for valid names and fields.
      * Requires scope: create:email_templates.
      *
-     * @param array $data
-     *      An array of data to use for the new email, including a valid `template`.
-     *      See docs link below for required fields.
+     * @param string $template - the template name to create
+     * @param string $from - email address the email should come from
+     * @param string $body - email body in the syntax indicated below
+     * @param boolean $enabled - is the email template enabled?
+     * @param string $subject - email subject
+     * @param string $syntax - email body syntax
+     * @param string $resultUrl - URL where a click should result
+     * @param int $urlLifetime - URL lifetime, in seconds
      *
      * @return mixed|string
      *
      * @throws \Exception - if a 200 response was not returned from the API.
      *
      * @link https://auth0.com/docs/api/management/v2#!/Email_Templates/post_email_templates
      */
-    public function create($data)
-    {
+    public function create(
+        $template,
+        $from,
+        $body = '',
+        $enabled = false,
+        $subject = '',
+        $syntax = 'liquid',
+        $resultUrl = '',
+        $urlLifetime = 0
+    ) {
+        // Required fields
+        $data = [
+            'template' => (string) $template,
+            'from' => (string) $from,
+            'body' => (string) $body,
```

Comment:
How does validation work here? And what happens since some of the params have a default value?

Reply:
Anything with a default value is not required by the API, only `template` and `from`. The object you POST needs to have those keys but they can be empty.

`$template` and `$from` are both required for the method (IOW: need to have something passed in) but nothing beyond that. Looking through the rest of the SDK, this is how validation is handled for other endpoints. I could check for `empty()` but it's a bit of an edge case if someone sees that the param is required but passes in an empty string or `null` or something. 

If it was empty, I'd have to `throw` and exception of some kind. Since this is a new method, I could do that even though none of the other ones do and make that a pattern to follow. If that template is empty, the HTTP client would throw an exception anyways so not a complete surprise. 

I'm happy with this as-is but also open to more discussion.

Reply:
I'm looking at the schema definition right now.. You shouldn't be setting values in the SDK if they are missing just because "need to have something passed in". Of course you'll need to pass something, that's why it's required. But let it fail gracefully! If the parameters are required, make the user manually set them, even if they want to set an empty string. The only exception I find for this is the `syntax` field which is there for backwards (read) compatibility and `liquid` should be used always as value when creating new templates. 

Think of receiving an email in your personal inbox which doesn't have a subject set or which doesn't have a body at all. Would that make sense to you as a user of that Application? Don't you want to make sure that users of this SDK use the API well?